### PR TITLE
[Merged by Bors] - fix(Util/FormatTable): escape bar characters in formatting tables for markdown

### PR DESCRIPTION
**Pull request by edegeltje**
This PR changes the `formatTable` function to make sure vertical bar (`|`) characters get escaped when formatting strings into markdown tables.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Could you add a test file in the tests directory using #guard_msgs?

**Comment by edegeltje** [comment]
i added a test, but in order to make sensible use of `#guard_msgs`, i added a macro to get rid of the call to `Repr` (which `#eval` calls). if this should be moved somewhere else, if i should just substitute it's contents at the single use, or if i should just cope with using `#eval`, please let me know.

**Comment by grunweg** [comment]
I think using a macro is fine. I tweaked the comment and test slightly. Thanks!
maintainer merge

File: test/format_table.lean [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/--
+info:
+| first           | second header   | third \| header |
+| :-------------- | :-------------: | --------------: |
+| item number one |    item two     |          item c |
+| the fourth item | the fourth item |  escape \| this |
+| align left      |  align center   |     align right |
+-/
+#guard_msgs in
+#logAsInfo "\n" ++ formatTable
```
Comment (eric-wieser) [inline test/format_table.lean:23]: I think `run_cmd Lean.logInfo` would be better hetet

Comment (edegeltje) [reply test/format_table.lean]: ah, that makes more sense! i thought it was weird that there wouldn't be a way to do this!

**Comment by kim-em** [comment]
bors merge

File: test/format_table.lean [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/--
+info:
+| first           | second header   | third \| header |
+| :-------------- | :-------------: | --------------: |
+| item number one |    item two     |          item c |
+| the fourth item | the fourth item |  escape \| this |
+| align left      |  align center   |     align right |
+-/
+#guard_msgs in
+run_cmd Lean.logInfo ("\n" ++ formatTable
```
Comment (eric-wieser) [inline test/format_table.lean:12]:
```suggestion
#eval Std.Format.text (formatTable
```
also works here

**Pull request closed without merging**